src/Autowhatever.js

```diff
@@ -98,19 +99,22 @@ export default class Autowhatever extends Component {
       const onClickFn = onClick ?
         event => onClick(event, { sectionIndex, itemIndex }) :
         noop;
+      const isFocused = sectionIndex === focusedSectionIndex && itemIndex === focusedItemIndex;
       const itemProps = {
         id: this.getItemId(sectionIndex, itemIndex),
         role: 'option',
-        ...theme(itemIndex, 'item', sectionIndex === focusedSectionIndex &&
-                                    itemIndex === focusedItemIndex &&
-                                    'itemFocused'),
+        ...theme(itemIndex, 'item', isFocused && 'itemFocused'),
         ...itemPropsObj,
         onMouseEnter: onMouseEnterFn,
         onMouseLeave: onMouseLeaveFn,
         onMouseDown: onMouseDownFn,
         onClick: onClickFn
       };
 
+      if (isFocused) {
+        itemProps.ref = 'focusedItem';
+      }
```

Comment:
The `const` keyword doesn't imply no mutations, it guarantees no change of memory address.  It just so happens that with primitive types the effect is _behaviorally_ no mutations.  Just my two cents, but I wouldn't get hung up over `const` being used this way.